core-java-modules/core-java-arrays-operations-advanced-2/src/test/java/com/baeldung/movezerototheend/MoveZeroesToTheEndOfAnArrayUnitTest.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.movezerototheend;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+public class MoveZeroesToTheEndOfAnArrayUnitTest {
+    static final int[] EXPECTED = new int[] { 1, 2, 3, 4, 0, 0 };
```

Comment:
Can we pick a slightly different example that has some unordered non-zero elements instead of the ordered sequence 1234? Just to emphasise that this algorithm doesn't do any kind of sorting, and only changes the position of the zeros

Reply:
I changed it to `42, 2, 3, 4, 0, 0`. So, sorting is excluded.